sources/packages/backend/workflow/test/2021-2022/parttime-assessment/awards-amounts/parttime-assessment-2021-2022-awards-amount-CSGD.e2e-spec.ts

```diff
@@ -0,0 +1,226 @@
+        .limitAwardCSPTAmount,
+    ).toBe(3600);
+    expect(
+      Math.min(
+        maxComparisonCalculation,
+        calculatedAssessment.variables.dmnPartTimeAwardAllowableLimits
+          .limitAwardCSPTAmount,
+      ),
+    ).toBe(maxComparisonCalculation);
+  });
+
+  it("Should determine federalAwardCSGDAmount when calculatedDataTotalFamilyIncome > limitAwardCSGDIncomeCap", async () => {
+    // Arrange
+    const assessmentConsolidatedData =
+      createFakeConsolidatedPartTimeData(PROGRAM_YEAR);
+    assessmentConsolidatedData.studentDataCRAReportedIncome = 35000;
+    assessmentConsolidatedData.studentDataRelationshipStatus = "married";
+    assessmentConsolidatedData.studentDataIsYourSpouseACanadianCitizen =
+      YesNoOptions.Yes;
+    assessmentConsolidatedData.partner1CRAReportedIncome = 35000;
+    assessmentConsolidatedData.studentDataDependants = [
+      createFakeStudentDependentEligible(
+        DependentEligibility.Eligible0To18YearsOld,
+      ),
+    ];
+    // Act
+    const calculatedAssessment = await executePartTimeAssessmentForProgramYear(
+      PROGRAM_YEAR,
+      assessmentConsolidatedData,
+    );
+    // Assert
+    // calculatedDataTotalFamilyIncome > limitAwardCSGDIncomeCap
+    expect(
+      calculatedAssessment.variables.calculatedDataTotalFamilyIncome,
+    ).toBeGreaterThan(
+      calculatedAssessment.variables.dmnPartTimeAwardFamilySizeVariables
+        .limitAwardCSGDIncomeCap,
+    );
+    const nestedCalculation =
+      calculatedAssessment.variables.dmnPartTimeAwardAllowableLimits
+        .limitAwardCSGD3OrMoreChildAmount -
+      (calculatedAssessment.variables.calculatedDataTotalFamilyIncome -
+        calculatedAssessment.variables.dmnPartTimeAwardFamilySizeVariables
+          .limitAwardCSGDIncomeCap) *
+        calculatedAssessment.variables.dmnPartTimeAwardFamilySizeVariables
+          .limitAwardCSGD3OrMoreChildSlope;
+    expect(nestedCalculation).toBeGreaterThan(0);
+    expect(Math.max(nestedCalculation, 0)).toBe(nestedCalculation);
+    const offeringWeeksAmount =
+      Math.max(nestedCalculation, 0) *
+      calculatedAssessment.variables.offeringWeeks;
+    const maxComparisonCalculation = Math.max(
+      offeringWeeksAmount,
+      calculatedAssessment.variables.dmnPartTimeAwardAllowableLimits
+        .limitAwardCSGD3OrMoreChildAmount,
+    );
+    expect(maxComparisonCalculation).toBe(offeringWeeksAmount);
+    expect(maxComparisonCalculation).toBe(1800);
+    expect(
+      calculatedAssessment.variables.dmnPartTimeAwardAllowableLimits
+        .limitAwardCSPTAmount,
+    ).toBe(3600);
+    expect(
+      Math.min(
+        maxComparisonCalculation,
+        calculatedAssessment.variables.dmnPartTimeAwardAllowableLimits
+          .limitAwardCSPTAmount,
+      ),
+    ).toBe(maxComparisonCalculation);
+  });
+});
+
+describe(`E2E Test Workflow parttime-assessment-${PROGRAM_YEAR}-awards-amount-CSGD for less than 3 dependants.`, () => {
```

Comment:
Usually we have one describe for a file, I am not against it but we is this okay for the other devs? @dheepak-aot @andrewsignori-aot @andrepestana-aot @sh16011993

Reply:
I would say we can follow the same here mainly because, the difference between 2 describe is the data for which we have been creating just more tests and we can follow the same.

Reply:
The describe should be only one and similar to the file name. The part about the dependants should be include in the `it()`.

Reply:
The test files are using only one describe now. Please have a look.